CI: Cilium L4LB NAT-46x64: array index out of bounds

https://github.com/cilium/cilium/runs/6038764931?check_suite_focus=true

```
+ kubectl -n kube-system delete pod -l k8s-app=cilium
pod "cilium-84b8f" deleted
++ kubectl -n kube-system get pod -l k8s-app=cilium '-o=jsonpath={.items[0].metadata.name}'
error: error executing jsonpath "{.items[0].metadata.name}": Error executing template: array index out of bounds: index 0, length 0. Printing more information for debugging the template:
	template was:
		{.items[0].metadata.name}
	object given to jsonpath engine was:
		map[string]interface {}{"apiVersion":"v1", "items":[]interface {}{}, "kind":"List", "metadata":map[string]interface {}{"resourceVersion":"", "selfLink":""}}


+ CILIUM_POD_NAME=
Error: Process completed with exit code 1.
```

--- a/test/nat46x64/test.sh
+++ b/test/nat46x64/test.sh
@@ -46,14 +46,14 @@ docker exec kind-worker /bin/sh -c 'apt-get update && apt-get install -y nginx &
 WORKER_IP6=$(docker exec kind-worker ip -o -6 a s eth0 | awk '{print $4}' | cut -d/ -f1 | head -n1)
 WORKER_IP4=$(docker exec kind-worker ip -o -4 a s eth0 | awk '{print $4}' | cut -d/ -f1 | head -n1)
 
-CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
-kubectl -n kube-system wait --for=condition=Ready pod "$CILIUM_POD_NAME" --timeout=5m
+kubectl -n kube-system rollout status ds/cilium --timeout=5m
 
 # NAT 4->6 test suite
 #####################
 
 LB_VIP="10.0.0.4"
 
+CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
 kubectl -n kube-system exec "${CILIUM_POD_NAME}" -- \
     cilium service update --id 1 --frontend "${LB_VIP}:80" --backends "[${WORKER_IP6}]:80" --k8s-node-port
 
@@ -77,11 +77,11 @@ helm upgrade cilium ../../install/kubernetes/cilium \
     --set loadBalancer.acceleration=native
 kubectl -n kube-system delete pod -l k8s-app=cilium
 
-CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
-kubectl -n kube-system wait --for=condition=Ready pod "$CILIUM_POD_NAME" --timeout=5m
+kubectl -n kube-system rollout status ds/cilium --timeout=5m
 
 # Check that restoration went fine. Note that we currently cannot do runtime test
 # as veth + XDP is broken when switching protocols. Needs something bare metal.
+CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
 SVC_AFTER=$(kubectl -n kube-system exec "${CILIUM_POD_NAME}" -- cilium service list)
 
 kubectl -n kube-system exec "${CILIUM_POD_NAME}" -- cilium bpf lb list
@@ -96,8 +96,7 @@ helm upgrade cilium ../../install/kubernetes/cilium \
     --set loadBalancer.acceleration=disabled
 kubectl -n kube-system delete pod -l k8s-app=cilium
 
-CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
-kubectl -n kube-system wait --for=condition=Ready pod "$CILIUM_POD_NAME" --timeout=5m
+kubectl -n kube-system rollout status ds/cilium --timeout=5m
 
 # Check that curl still works after restore
 for i in $(seq 1 10); do
@@ -112,8 +111,7 @@ helm upgrade cilium ../../install/kubernetes/cilium \
     --set loadBalancer.algorithm=random
 kubectl -n kube-system delete pod -l k8s-app=cilium
 
-CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
-kubectl -n kube-system wait --for=condition=Ready pod "$CILIUM_POD_NAME" --timeout=5m
+kubectl -n kube-system rollout status ds/cilium --timeout=5m
 
 # Check that curl also works for random selection
 for i in $(seq 1 10); do
@@ -124,6 +122,7 @@ done
 
 LB_ALT="fd00:dead:beef:15:bad::1"
 
+CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
 kubectl -n kube-system exec "${CILIUM_POD_NAME}" -- \
     cilium service update --id 2 --frontend "[${LB_ALT}]:80" --backends "[${WORKER_IP6}]:80" --k8s-node-port
 
@@ -153,8 +152,7 @@ helm upgrade cilium ../../install/kubernetes/cilium \
     --set loadBalancer.algorithm=maglev
 kubectl -n kube-system delete pod -l k8s-app=cilium
 
-CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
-kubectl -n kube-system wait --for=condition=Ready pod "$CILIUM_POD_NAME" --timeout=5m
+kubectl -n kube-system rollout status ds/cilium --timeout=5m
 
 # Issue 10 requests to LB1
 for i in $(seq 1 10); do
@@ -166,6 +164,7 @@ for i in $(seq 1 10); do
     curl -o /dev/null "[${LB_ALT}]:80"
 done
 
+CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
 kubectl -n kube-system exec "${CILIUM_POD_NAME}" -- cilium service delete 1
 kubectl -n kube-system exec "${CILIUM_POD_NAME}" -- cilium service delete 2
 
@@ -197,11 +196,11 @@ helm upgrade cilium ../../install/kubernetes/cilium \
     --set loadBalancer.acceleration=native
 kubectl -n kube-system delete pod -l k8s-app=cilium
 
-CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
-kubectl -n kube-system wait --for=condition=Ready pod "$CILIUM_POD_NAME" --timeout=5m
+kubectl -n kube-system rollout status ds/cilium --timeout=5m
 
 # Check that restoration went fine. Note that we currently cannot do runtime test
 # as veth + XDP is broken when switching protocols. Needs something bare metal.
+CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
 SVC_AFTER=$(kubectl -n kube-system exec "${CILIUM_POD_NAME}" -- cilium service list)
 
 kubectl -n kube-system exec "${CILIUM_POD_NAME}" -- cilium bpf lb list
@@ -216,8 +215,7 @@ helm upgrade cilium ../../install/kubernetes/cilium \
     --set loadBalancer.acceleration=disabled
 kubectl -n kube-system delete pod -l k8s-app=cilium
 
-CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
-kubectl -n kube-system wait --for=condition=Ready pod "$CILIUM_POD_NAME" --timeout=5m
+kubectl -n kube-system rollout status ds/cilium --timeout=5m
 
 # Check that curl still works after restore
 for i in $(seq 1 10); do
@@ -232,8 +230,7 @@ helm upgrade cilium ../../install/kubernetes/cilium \
     --set loadBalancer.algorithm=random
 kubectl -n kube-system delete pod -l k8s-app=cilium
 
-CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
-kubectl -n kube-system wait --for=condition=Ready pod "$CILIUM_POD_NAME" --timeout=5m
+kubectl -n kube-system rollout status ds/cilium --timeout=5m
 
 # Check that curl also works for random selection
 for i in $(seq 1 10); do
@@ -244,6 +241,7 @@ done
 
 LB_ALT="10.0.0.8"
 
+CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
 kubectl -n kube-system exec "${CILIUM_POD_NAME}" -- \
     cilium service update --id 2 --frontend "${LB_ALT}:80" --backends "${WORKER_IP4}:80" --k8s-node-port
 
@@ -273,8 +271,7 @@ helm upgrade cilium ../../install/kubernetes/cilium \
     --set loadBalancer.algorithm=maglev
 kubectl -n kube-system delete pod -l k8s-app=cilium
 
-CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
-kubectl -n kube-system wait --for=condition=Ready pod "$CILIUM_POD_NAME" --timeout=5m
+kubectl -n kube-system rollout status ds/cilium --timeout=5m
 
 # Issue 10 requests to LB1
 for i in $(seq 1 10); do
@@ -286,6 +283,7 @@ for i in $(seq 1 10); do
     curl -o /dev/null "${LB_ALT}:80"
 done
 
+CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
 kubectl -n kube-system exec "${CILIUM_POD_NAME}" -- cilium service delete 1
 kubectl -n kube-system exec "${CILIUM_POD_NAME}" -- cilium service delete 2
 
@@ -302,10 +300,10 @@ helm upgrade cilium ../../install/kubernetes/cilium \
     --set loadBalancer.acceleration=native
 kubectl -n kube-system delete pod -l k8s-app=cilium
 
-CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
-kubectl -n kube-system wait --for=condition=Ready pod "$CILIUM_POD_NAME" --timeout=5m
+kubectl -n kube-system rollout status ds/cilium --timeout=5m
 
 # Trigger recompilation with 32 IPv4 filter masks
+CILIUM_POD_NAME=$(kubectl -n kube-system get pod -l k8s-app=cilium -o=jsonpath='{.items[0].metadata.name}')
 kubectl -n kube-system exec "${CILIUM_POD_NAME}" -- \
     cilium recorder update --id 1 --caplen 100 \
         --filters="2.2.2.2/0 0 1.1.1.1/32 80 TCP,\

